[UE5.3] Merge pull request #271 from mcottontensor/fix_fake_mouse

mcottontensor opened the pull request:
# Backport

This will backport the following commits from `master` to `UE5.3`:
 - [Merge pull request #271 from mcottontensor/fix_fake_mouse](https://github.com/EpicGamesExt/PixelStreamingInfrastructure/pull/271)

<!--- Backport version: 9.5.1 -->

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

jibranabsarulislam commented:
please feel free to provide your feedback and suggestions, especially with the tests; they're a top weakness of mine

lukehb commented on Frontend/library/src/Util/EventEmitter.ts:
```diff
@@ -457,6 +487,9 @@ export type PixelStreamingEvent =
     | DataChannelErrorEvent
     | VideoInitializedEvent
     | StreamLoadingEvent
+    | StreamConnectEvent
```
Considering when this event fire does, I would prefer the name `StreamPreConnect`

lukehb commented on Frontend/library/src/Util/EventEmitter.ts:
```diff
@@ -457,6 +487,9 @@ export type PixelStreamingEvent =
     | StreamLoadingEvent
+    | StreamConnectEvent
+    | StreamReconnectEvent
+    | StreamDisconnectEvent
```
Similiar here, `StreamPreDisconnect`

lukehb reviewed (changes requested):
Looks good, thanks for this.

Can I ask you change the events to StreamPreConnect and StreamPreDisconnect - sorry that wasn't captured in the issue originally.

jibranabsarulislam commented:
> Looks good, thanks for this.
> 
> Can I ask you change the events to StreamPreConnect and StreamPreDisconnect - sorry that wasn't captured in the issue originally.

no worries, got those changes a small refactor pushed. Thank you!

If possible, can I be tagged on early discussion of these events' implementation? i'm interested in learning more about the connection/networking within pixelstreaming

jibranabsarulislam commented:
no worries—pushed!
thank you @lukehb

jibranabsarulislam commented on Frontend/library/src/PixelStreaming/PixelStreaming.test.ts:
```diff
@@ -572,4 +572,52 @@ describe('PixelStreaming', () => {
+        pixelStreaming.disconnect();
+        pixelStreaming.connect();
+
+        expect(streamReconnectSpy).toHaveBeenCalled();
```
ah, my mistake, thank you for fix!

mcottontensor committed:
Faking mouse with touch is now once again false by default. Added the option to the settings panel

(cherry picked from commit 58417776b329daf3de329b2f01f87f845738ffed)

# Conflicts:
#	Frontend/library/src/Config/Config.ts
#	Frontend/library/src/PixelStreaming/PixelStreaming.ts
#	Frontend/ui-library/src/Config/ConfigUI.ts
